src/FunctionalLimitedCollection.js

```diff
@@ -0,0 +1,128 @@
+import {FunctionalArrayCollection} from './FunctionalArrayCollection'
+
+/**
+ * Throw Error if add new item to collection and collection
+ * hold too many items.
+ *
+ * @since [*next-version*]
+ *
+ * @type {string}
+ */
+export const FLC_OVERFLOW_THROW = 'throw';
+
+/**
+ * Start adding item to the collection from start if
+ * there are to many items in the collection.
+ *
+ * @since [*next-version*]
+ *
+ * @type {string}
+ */
+export const FLC_OVERFLOW_START = 'start';
+
+/**
+ * Functional limited collection that works with arrays under the hood
+ * and can hold maximal amount of items.
+ *
+ * @since [*next-version*]
+ */
+export class FunctionalLimitedCollection extends FunctionalArrayCollection {
+
+    /**
+     * @var {Number} Count of maximal items count in collection.
+     *
+     * @since [*next-version*]
+     */
+    _limit;
+
+    /**
+     * @var {String} Define how to deal with items overflow.
+     *
+     * @since [*next-version*]
+     */
+    _overflowMode;
+
+    /**
+     * @var {Number} Index of last added items.
+     *
+     * @since [*next-version*]
+     */
+    _lastAddedIndex;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param {Function} getItems The function used to retrieve the state of the item set.
+     * @param {Function} setItems The function used to commit the new state of the item set.
+     * @param {Function} keyGetter The function used to get item's key.
+     * @param {Number} limit Count of maximal items count in collection.
+     * @param {String} overflowMode Define how to deal with items overflow.
+     */
+    constructor (getItems, setItems, keyGetter, limit, overflowMode) {
+        if (!limit) {
+            throw new Error(`Collection limit must be provided`)
+        }
+
+        if (!overflowMode) {
+            throw new Error(`Overflow mode for collection must be provided`)
```

Comment:
It would be great if there was also a list of allowed modes, and you would throw here if the mode is invalid.